ledger/participant-integration-api/src/main/scala/platform/store/backend/StorageBackend.scala

```diff
@@ -55,13 +94,299 @@ trait StorageBackend[DB_BATCH] {
+      endInclusiveSeq: Long,
+      endInclusiveOffset: Offset,
+      party: Ref.Party,
+      templateIds: Set[Ref.Identifier],
+      limit: Option[Int],
+      fetchSizeHint: Option[Int],
+  )(connection: Connection): Vector[EventsTable.Entry[Raw.FlatEvent]]
+  def activeContractsEventsOnlyWildcardParties(
+      startExclusive: Long,
+      endInclusiveSeq: Long,
+      endInclusiveOffset: Offset,
+      parties: Set[Ref.Party],
+      limit: Option[Int],
+      fetchSizeHint: Option[Int],
+  )(connection: Connection): Vector[EventsTable.Entry[Raw.FlatEvent]]
+  def activeContractsEventsSameTemplates(
+      startExclusive: Long,
+      endInclusiveSeq: Long,
+      endInclusiveOffset: Offset,
+      parties: Set[Ref.Party],
+      templateIds: Set[Ref.Identifier],
+      limit: Option[Int],
+      fetchSizeHint: Option[Int],
+  )(connection: Connection): Vector[EventsTable.Entry[Raw.FlatEvent]]
+  def activeContractsEventsMixedTemplates(
+      startExclusive: Long,
+      endInclusiveSeq: Long,
+      endInclusiveOffset: Offset,
+      partiesAndTemplateIds: Set[(Ref.Party, Ref.Identifier)],
+      limit: Option[Int],
+      fetchSizeHint: Option[Int],
+  )(connection: Connection): Vector[EventsTable.Entry[Raw.FlatEvent]]
+  def activeContractsEventsMixedTemplatesWithWildcardParties(
+      startExclusive: Long,
+      endInclusiveSeq: Long,
+      endInclusiveOffset: Offset,
+      partiesAndTemplateIds: Set[(Ref.Party, Ref.Identifier)],
+      wildcardParties: Set[Ref.Party],
+      limit: Option[Int],
+      fetchSizeHint: Option[Int],
+  )(connection: Connection): Vector[EventsTable.Entry[Raw.FlatEvent]]
+  def flatTransactionSingleParty(
+      transactionId: TransactionId,
+      requestingParty: Ref.Party,
+  )(connection: Connection): Vector[EventsTable.Entry[Raw.FlatEvent]]
+  def flatTransactionMultiParty(
+      transactionId: TransactionId,
+      requestingParties: Set[Ref.Party],
+  )(connection: Connection): Vector[EventsTable.Entry[Raw.FlatEvent]]
+  def transactionTreeSingleParty(
+      transactionId: TransactionId,
+      requestingParty: Ref.Party,
+  )(connection: Connection): Vector[EventsTable.Entry[Raw.TreeEvent]]
+  def transactionTreeMultiParty(
+      transactionId: TransactionId,
+      requestingParties: Set[Ref.Party],
+  )(connection: Connection): Vector[EventsTable.Entry[Raw.TreeEvent]]
+  def transactionTreeEventsSingleParty(
+      startExclusive: Long,
+      endInclusive: Long,
+      requestingParty: Ref.Party,
+      limit: Option[Int],
+      fetchSizeHint: Option[Int],
+  )(connection: Connection): Vector[EventsTable.Entry[Raw.TreeEvent]]
+  def transactionTreeEventsMultiParty(
+      startExclusive: Long,
+      endInclusive: Long,
+      requestingParties: Set[Ref.Party],
+      limit: Option[Int],
+      fetchSizeHint: Option[Int],
+  )(connection: Connection): Vector[EventsTable.Entry[Raw.TreeEvent]]
+  def maxEventSeqIdForOffset(offset: Offset)(connection: Connection): Option[Long]
```

Comment:
Can you move this method to the parameters-related trait?

Reply:
not really, this has nothing to do with parameters, it is a search in the events view

Reply:
Right